tests/common/util.rs

```diff
@@ -1191,26 +1221,613 @@ impl UCommand {
+
+        Ok(CmdResult {
+            bin_path,
+            util_name,
+            tmpd,
+            code: output.status.code(),
+            success: output.status.success(),
+            stdout: output.stdout,
+            stderr: output.stderr,
+        })
+    }
+
+    /// Wait for the child process to terminate and return an instance of [`Output`]. Joins
+    /// with the thread created by [`UChild::pipe_in`] if any.
+    pub fn wait_with_output(mut self) -> io::Result<Output> {
+        let mut output = self.raw.wait_with_output()?;
+
+        if let Some(join_handle) = self.join_handle.take() {
+            join_handle
+                .join()
+                .expect("Error joining with the piping stdin thread")
+                .unwrap();
+        };
+
+        if let Some(stdout) = self.captured_stdout.as_mut() {
+            output.stdout = stdout.output_bytes();
+        }
+        if let Some(stderr) = self.captured_stderr.as_mut() {
+            output.stderr = stderr.output_bytes();
+        }
+
+        Ok(output)
+    }
+
+    /// Reads, consumes and returns the output as [`String`] from [`Child`]'s stdout. See also
+    /// [`UChild::stdout_bytes] for side effects.
+    pub fn stdout(&mut self) -> String {
+        String::from_utf8(self.stdout_bytes()).unwrap()
+    }
+
+    /// Reads and returns all child's output in stdout as String. Note, that a subsequent call
+    /// of any of these functions
+    ///
+    /// * [`UChild::stdout`]
+    /// * [`UChild::stdout_bytes`]
+    /// * [`UChild::stdout_exact_bytes`]
+    ///
+    /// will operate on the subsequent output of the child process.
+    pub fn stdout_all(&mut self) -> String {
+        String::from_utf8(self.stdout_all_bytes()).unwrap()
+    }
+
+    /// Reads, consumes and returns the output as bytes from [`Child`]'s stdout. Each subsequent
+    /// call to any of the functions below will operate on the subsequent output of the child
+    /// process:
+    ///
+    /// * [`UChild::stdout`]
+    /// * [`UChild::stdout_exact_bytes`]
+    /// * and the call to itself [`UChild::stdout_bytes`]
+    pub fn stdout_bytes(&mut self) -> Vec<u8> {
+        match self.captured_stdout.as_mut() {
+            Some(output) => output.output_bytes(),
+            None if self.raw.stdout.is_some() => {
+                let mut buffer: Vec<u8> = vec![];
+                let stdout = self.raw.stdout.as_mut().unwrap();
+                stdout.read_to_end(&mut buffer).unwrap();
+                buffer
+            }
+            None => vec![],
+        }
+    }
+
+    /// Reads and returns all output from start of the child process until now. Each subsequent call
+    /// of any of the methods below will operate on the subsequent output of the child process. Note
+    /// that this function will work like [`UChild::stdout_bytes`] if the output wasn't captured
+    /// with [`CapturedOutput`], for example if [`UCommand::set_stdout`] was used with
+    /// [`Stdio::piped`].
```

Comment:
Two things about this behaviour.
1. If it would be the same, I would assume it's user error and we should maybe panic if this is called without a captured stdout, but I'm not sure.
2. If you do keep it, the behaviour should be documented on `stdout_all` and `stderr_all` as well.

Reply:
Let's panic then. It's definitely an usage error.